web/src/components/pages/contribution/listen/listen.tsx

```diff
@@ -134,19 +134,19 @@ class ListenPage extends React.Component<Props, State> {
     trackListening('listen', this.props.locale);
   };
 
-  private vote = (isValid: boolean) => {
+  private vote = async (isValid: boolean) => {
     const { clips } = this.state;
+    const clipIndex = this.getClipIndex();
+
+    this.stop();
+    await this.props.vote(isValid, this.state.clips[this.getClipIndex()].id);
     const {
       firstContribute,
       hasAchieved,
       addAchievement,
       api,
       firstStreak,
     } = this.props;
```

Comment:
Let's move this back to where it was before so that the diff is simpler. As far as I can tell only two lines actually changed (adding async / await).

Reply:
please don‘t. This line won’t change after this.props gets updated. For example, firstContribute will always be false.

It has to be where vote() gets called and returned.

Reply:
Forgot to mention, this.props.vote() updates attributes such as firstContribute, hasAchieved...

I had subsequent commits to change these attributes’ names to something like showFirstContributeToast, hasEarnedxxxBonus etc. Also fix the bug about multiple same_session toasts AFTER challenge’s end date.
I will put them in another PR tomorrow.

Reply:
Oh! Surprising - thanks for the extra info. With that extra information I'd say this is good to go.